src/lib/components/market/response_render_results_status.ts

```diff
@@ -0,0 +1,155 @@
+function overrideOnResponseRenderResults(proto: any) {
+    const originalMethod = proto.OnResponseRenderResults;
+
+    proto.OnResponseRenderResults = function(responseClass: any) {
+        const success = (responseClass && responseClass.responseJSON && responseClass.responseJSON.success) || false;
+        if (!success) {
+            paginationBox.showError();
+        } else {
+            paginationBox.hide();
+        }
+
+        originalMethod.call(this, responseClass);
+    };
+}
+
+function overrideGoToPage(proto: any) {
+    const originalMethodGoToPage = proto.GoToPage;
+
+    proto.GoToPage = function(page: number) {
+        paginationBox.showPending();
+        originalMethodGoToPage.call(this, page);
+    };
+}
+
+export function initResponseRenderResultsStatus() {
+    const id = setInterval(() => {
+        const searchResultsObject = g_oSearchResults;
+        if (!searchResultsObject) return;
+
+        const proto = Object.getPrototypeOf(searchResultsObject);
+
+        overrideOnResponseRenderResults(proto);
+        overrideGoToPage(proto);
+
+        clearInterval(id);
+    }, 1000);
+}
+
+class PaginationFailedBox extends HTMLElement {
+    private container: HTMLDivElement;
+    private progressBar: HTMLDivElement;
+    private messageDiv: HTMLDivElement;
+    private hideTimeout: number | null = null;
+    private timeoutDuration: number = 1;
```

Comment:
This has no clear units, for example, it seems to represent 5s increments for show pending and seconds for error.

You should put the units in the variable name (ie. `timeoutMs`) and use it consistently.